pkg/operator/controller/nodeobservability/rbac.go

```diff
@@ -3,163 +3,81 @@ package nodeobservabilitycontroller
 import (
 	"context"
 	"fmt"
+	"reflect"
 
 	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
 	v1alpha1 "github.com/openshift/node-observability-operator/api/v1alpha1"
 )
 
 const (
-	apiGroup                     = "rbac.authorization.k8s.io"
-	clusterRoleName              = "node-observability-cr"
-	clusterRole                  = "ClusterRole"
-	clusterRoleBindingName       = "node-observability-crb"
-	serviceAccount               = "ServiceAccount"
-	get                          = "get"
-	create                       = "create"
-	list                         = "list"
-	nodes                        = "nodes"
-	nodesProxy                   = "nodes/proxy"
-	pods                         = "pods"
-	urlStatus                    = "/node-observability-status"
-	urlPprof                     = "/node-observability-pprof"
-	secGroup                     = "security.openshift.io"
-	authnGroup                   = "authentication.k8s.io"
-	authzGroup                   = "authorization.k8s.io"
-	secResource                  = "securitycontextconstraints"
-	secResourceName              = "node-observability-scc"
-	tokenreviewsResource         = "tokenreviews"
-	subjectaccessreviewsResource = "subjectaccessreviews"
-	use                          = "use"
-)
 
-// ensureClusterRole ensures that the clusterrole exists
-// Returns a Boolean value indicating whether it exists, a pointer to
-// cluster role and an error when relevant
-func (r *NodeObservabilityReconciler) ensureClusterRole(ctx context.Context, nodeObs *v1alpha1.NodeObservability) (bool, *rbacv1.ClusterRole, error) {
-	nameSpace := types.NamespacedName{Name: clusterRoleName}
-	desired := r.desiredClusterRole(nodeObs)
-	exist, current, err := r.currentClusterRole(ctx, nameSpace)
-	if err != nil {
-		return false, nil, fmt.Errorf("failed to get ClusterRole: %w", err)
-	}
-	if !exist {
-		if err := r.createClusterRole(ctx, desired); err != nil {
-			return false, nil, err
-		}
-		return r.currentClusterRole(ctx, nameSpace)
-	}
-	return true, current, err
-}
+	// TODO: standardize resource names to keep them compatible
+	clusterRoleName        = "node-observability-operator-controller-role"
+	clusterRoleBindingName = "node-observability-operator-controller-role-binding"
+)
```

Comment:
Currently dummy names are assigned. Need to look into how to name these resources. Currently has a lot of inconsistency, 
- the SA that is mounted on the nob agent is called `node-observability-sa`
- the clusterrole assigned to the agent used to be called `node-observability-cr`
- the clusterrolebinding used to be called `node-observability-crb` 
Even though all of these are created for the operand.